Source: Upstream
Upstream: Yes
Reason: Fix build failure with recent phonon (?)

From 91ed779fd1c83d8d344dd0f851282e5d58701cd4 Mon Sep 17 00:00:00 2001
From: rdieter <rdieter@283d02a7-25f6-0310-bc7c-ecb5cbfe19da>
Date: Fri, 12 Nov 2010 20:48:59 +0000
Subject: [PATCH] a minimal variant of trunk's r1173672: ifdef out some dup/conflicting
 Q_DECLARE_METATYPE's to fix build with newer phonon.

git-svn-id: svn+ssh://svn.kde.org/home/kde/branches/KDE/4.5/kdebase/runtime@1196219 283d02a7-25f6-0310-bc7c-ecb5cbfe19da
---
 phonon/kded-module/phononserver.cpp       |    3 +++
 phonon/platform_kde/devicelisting.cpp     |    3 +++
 phonon/platform_kde/kdeplatformplugin.cpp |    2 ++
 3 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/phonon/kded-module/phononserver.cpp b/phonon/kded-module/phononserver.cpp
index 86f380a..8d6fb40 100644
--- a/phonon/kded-module/phononserver.cpp
+++ b/phonon/kded-module/phononserver.cpp
@@ -18,6 +18,7 @@
 
 */
 
+#include "phononnamespace.h"
 #include "phononserver.h"
 #include "audiodevice.h"
 
@@ -56,7 +57,9 @@ K_PLUGIN_FACTORY(PhononServerFactory,
 K_EXPORT_PLUGIN(PhononServerFactory("phononserver"))
 
 typedef QList<QPair<QByteArray, QString> > PhononDeviceAccessList;
+#if (PHONON_VERSION < PHONON_VERSION_CHECK(4, 4, 3))
 Q_DECLARE_METATYPE(PhononDeviceAccessList)
+#endif
 
 PhononServer::PhononServer(QObject *parent, const QList<QVariant> &)
     : KDEDModule(parent),
diff --git a/phonon/platform_kde/devicelisting.cpp b/phonon/platform_kde/devicelisting.cpp
index 74c4bd3..e2aa5c3 100644
--- a/phonon/platform_kde/devicelisting.cpp
+++ b/phonon/platform_kde/devicelisting.cpp
@@ -19,6 +19,7 @@
 */
 
 #include "devicelisting.h"
+#include "phononnamespace.h"
 
 #include <QtCore/QFile>
 #include <QtDBus/QDBusReply>
@@ -35,7 +36,9 @@
 #endif // HAVE_ALSA_ASOUNDLIB_H
 
 typedef QList<QPair<QByteArray, QString> > PhononDeviceAccessList;
+#if (PHONON_VERSION < PHONON_VERSION_CHECK(4, 4, 3))
 Q_DECLARE_METATYPE(PhononDeviceAccessList)
+#endif
 
 static void installAlsaPhononDeviceHandle()
 {
diff --git a/phonon/platform_kde/kdeplatformplugin.cpp b/phonon/platform_kde/kdeplatformplugin.cpp
index 6ea9156..bbb41e6 100644
--- a/phonon/platform_kde/kdeplatformplugin.cpp
+++ b/phonon/platform_kde/kdeplatformplugin.cpp
@@ -43,9 +43,11 @@
 
 typedef QPair<QByteArray, QString> PhononDeviceAccess;
 typedef QList<PhononDeviceAccess> PhononDeviceAccessList;
+#if (PHONON_VERSION < PHONON_VERSION_CHECK(4, 4, 3))
 #ifndef KDE_USE_FINAL
 Q_DECLARE_METATYPE(PhononDeviceAccessList)
 #endif
+#endif
 namespace Phonon
 {
 
-- 
1.7.3.2

